firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -522,13 +522,25 @@ static long parseTokenExpirationTimestamp(String expiresIn) {
         : Long.parseLong(expiresIn.substring(0, expiresIn.length() - 1));
   }
 
-  private static void logFisCommunicationError(HttpURLConnection conn) {
+  private static void logFisCommunicationError(
+      HttpURLConnection conn,
+      @Nullable String appId,
+      @NonNull String apiKey,
+      @NonNull String projectId) {
     String logString = readErrorResponse(conn);
     if (!TextUtils.isEmpty(logString)) {
       Log.w(FIS_TAG, logString);
+      Log.w(FIS_TAG, availableFirebaseOptions(appId, apiKey, projectId));
     }
   }
 
+  private static String availableFirebaseOptions(
+      @Nullable String appId, @NonNull String apiKey, @NonNull String projectId) {
+    return String.format(
+        "Firebase options used while communicating with Firebase server APIs: %s, %s, %s",
+        apiKey, projectId, TextUtils.isEmpty(appId) ? "" : appId);
```

Comment:
It's not that I feel strongly about it, but this is not really necessary.

How about this compromise:
```
return String.format(
        "Firebase options used while communicating with Firebase server APIs: %s, %s%s",
        apiKey, projectId, TextUtils.isEmpty(appId) ? "" : ", " + appId);
```